plugins/poll/assets/javascripts/components/poll-breakdown-chart.js.es6

```diff
@@ -0,0 +1,175 @@
+import I18n from "I18n";
+import Component from "@ember/component";
+import { htmlSafe } from "@ember/template";
+import discourseComputed from "discourse-common/utils/decorators";
+import { PIE_CHART_TYPE } from "../controllers/poll-ui-builder";
+import { getColors } from "../lib/chart-colors";
+
+// args: options, group, displayMode, highlightedOption, setHighlightedOption
+export default Component.extend({
+  classNames: "poll-breakdown-chart-container",
+  optionToSlice: {},
+  previousHighlightedSliceIndex: null,
+  previousDisplayMode: null,
+
+  didInsertElement() {
+    this._super(...arguments);
+
+    const canvas = this.element.querySelector("canvas");
+    this.set(
+      "chart",
+      new window.Chart(canvas.getContext("2d"), this.chartConfig)
+    );
+  },
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+
+    if (this.chart) {
+      this._updateDisplayMode();
+      this._updateHighlight();
+    }
+  },
+
+  willDestroy() {
+    this._super(...arguments);
+
+    if (this.chart) {
+      this.chart.destroy();
+    }
+  },
+
+  @discourseComputed("options")
```

Comment:
can we use this here https://api.emberjs.com/ember/release/functions/@ember%2Fobject%2Fcomputed/mapBy ?